src/page_managers/match_manager.py

```diff
@@ -62,3 +62,131 @@ def generate_input_section(self) -> list[list, list]:
+                )
+
+            with auto_cycles_by_level_col:
+                auto_cycles_by_level = [[self.calculated_stats.cycles_by_height_per_match(team, Queries.AUTO_GRID, Queries.HIGH).mean() for team in team_numbers], [self.calculated_stats.cycles_by_height_per_match(team, Queries.AUTO_GRID, Queries.MID).mean() for team in team_numbers], [self.calculated_stats.cycles_by_height_per_match(team, Queries.AUTO_GRID, Queries.LOW).mean() for team in team_numbers]]
+                st.plotly_chart(
+                    stacked_bar_graph(
+                        x=team_numbers,
+                        y=auto_cycles_by_level,
+                        x_axis_label="Teams",
+                        y_axis_label=[
+                            "High","Mid", "Low"
+                        ],
+                        y_axis_title="Auto Game Pieces by Level",
+                    ),use_container_width=True
+                )
+
+            # Graph for auto charge station psuccess_rate
+            auto_engage_success_rate_per_team = [[self.calculated_stats.auto_engage_success_rate(team), self.calculated_stats.auto_total_attempted_charge(team), self.calculated_stats.auto_total_successful_charge(team)] for team in team_numbers]
+
+            st.plotly_chart(
+                    bar_graph(
+                        x=team_numbers,
+                        y=auto_engage_success_rate_per_team,
+                        x_axis_label="Teams",
+                        y_axis_label="Auto Engage Success Rate",
+                        hover_data=["Attempts", "Successful Engages"]
+                    ),use_container_width=True
+                )
+            
+
+        # Teleop + endgame graphs
+        with teleop_graphs_tab:
+            st.write("#### Teleop")
+
+            # Graph for teleop cycles over time
+            teleop_cycles_over_time_per_team =  [self.calculated_stats.points_contributed_by_match(team, Queries.TELEOP_GRID) if display_points else self.calculated_stats.cycles_by_match(team, Queries.TELEOP_GRID) for team in team_numbers]
+
+            st.plotly_chart(multi_line_graph(
+                    x=teams_data[0][Queries.MATCH_KEY],
+                    y=teleop_cycles_over_time_per_team,
+                    x_axis_label="Match Key",
+                    y_axis_label=team_numbers,
+                    y_axis_title="Teleop Point Total" if display_points else "# of Teleop Cycles"
+                    ),
+                    use_container_width=True
+            )
+                
+
+            teleop_cycles_cone_vs_cube_col, teleop_cycles_by_level_col = st.columns(2)
+
+            with teleop_cycles_cone_vs_cube_col:
+                teleop_cycles_cone_vs_cube = [[self.calculated_stats.cycles_by_game_piece_per_match(team, Queries.TELEOP_CONES).mean() for team in team_numbers], [self.calculated_stats.cycles_by_game_piece_per_match(team, Queries.TELEOP_CUBES).mean() for team in team_numbers]]
```

Comment:
See previous comment about avoiding putting big list comprehensions in one line, should probably break this into mulitple lines so it's easier to read even if it's still one big list comprehension